src/pages/Product.tsx

```diff
@@ -3,17 +3,46 @@ import * as React from 'react';
 import { ProductInformation } from '../sections/';
 import { AllProductDescription, SimpleBreadcrumbs } from '../components';
 
-function Product() {
+import { fetchJewelryById } from '../redux/reducers/jewelry/action';
+import { IJewelryItem } from '../redux/reducers/jewelry/type';
+
+import * as H from 'history';
+
+interface Product {
```

Comment:
For code to be cleaner - move types to a separate files.
If type can be re-used in different components - move to a global types for the project.